pandas/formats/format.py

```diff
@@ -489,30 +489,48 @@ def _to_str_columns(self):
         str_index = self._get_formatted_index(frame)
         str_columns = self._get_formatted_column_labels(frame)
 
-        if self.header:
+        has_aliases = isinstance(self.header, (tuple, list, np.ndarray, Index))
```

Comment:
Just want to mention that the original reason why I wrote it in this "unpythonic" way was to have consistency with ``CSVFormatter``.
It really seems as if ``CSVFormatter`` also could be refurbished.
But for the time being I just keep it in mind for when they are both transformed to using the Mixin class you mentioned.